zk/hermez_db/db.go

```diff
@@ -849,3 +875,179 @@ func (db *HermezDbReader) GetBlockInfoRoot(blockNumber uint64) (common.Hash, err
 	res := common.BytesToHash(data)
 	return res, nil
 }
+
+func (db *HermezDbReader) GetWitnessByBatchNo(batchNo uint64) ([]byte, error) {
+	w, err := ReadChunks(db.tx, BATCH_WITNESS, Uint64ToBytes(batchNo))
+	if err != nil {
+		return nil, err
+	}
+
+	return w, nil
+}
+
+func (db *HermezDbReader) GetLatestStoredWitnessBatchNo() (uint64, error) {
+	cursor, err := db.tx.Cursor(BATCH_WITNESS)
+	if err != nil {
+		return 0, err
+	}
+
+	k, _, err := cursor.Last()
+	if err != nil {
+		return 0, err
+	}
+
+	return batchNoFromWitnessKey(k)
+}
+
+func (db *HermezDbReader) GetOldestStoredWitnessBatch() (uint64, error) {
+	cursor, err := db.tx.Cursor(BATCH_WITNESS)
+	if err != nil {
+		return 0, err
+	}
+
+	key, _, err := cursor.First()
+	if err != nil {
+		return 0, err
+	}
+	// There is no data stored in the batch witness table
+	if len(key) == 0 {
+		return 0, nil
+	}
+
+	if len(key) < 8 {
+		return 0, fmt.Errorf("invalid key, byte slice smaller 8 cannot contain uint64")
+	}
+
+	// Removes `_chuck_ + chunk_id` that we add while storing it in db
+	batch := key[0:9]
```

Comment:
One thing which I saw while trying a similar approach is that the size is variable based on how big the number is. If it  can be accumulated in less bytes, it won't be 8 bytes long.

Reply:
It wil be you can refer(https://github.com/0xPolygonHermez/cdk-erigon/blob/zkevm/zk/hermez_db/utils.go#L18), also I found out there is a function already in the same file which convert bytesToUint64 so I can remove the function, but the logic there is pretty much the same.